library/analytics/src/main/java/org/cru/godtools/analytics/firebase/FirebaseAnalyticsService.kt

```diff
@@ -86,7 +95,8 @@ class FirebaseAnalyticsService @MainThread @Inject internal constructor(
 
     init {
         oktaUserProfileProvider.userInfoFlow()
-            .mapNotNull { it?.ssoGuid }
+            .map { it?.ssoGuid }
+            .distinctUntilChanged()
```

Comment:
Does this just listen for change and once there is it will update the value? For example, logging out, the values is changed, so we set the userId to ```null```.

Reply:
it's a small optimization.

if you have a stream of values that looks like: `null`, `a`, `a`, `b`, `null`, `null` it would filter out duplicate values that occur and wouldn't change anything. it would become: `null`, `a`, `b`, `null`.

Reply:
the `distinctUntilChanged()` flow operator will probably go away when we start reporting additional user profile values, but with just a single value it makes sense for now.